File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.js
var addSpaces = function(s, spaces) {
    const result = [];
    const spaceSet = new Set(spaces);

    for (let i = 0; i < s.length; i++) {
        if (spaceSet.has(i)) {
            result.push(" ");
        }
        result.push(s[i]);
    }

    return result.join("");
};

File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.py
class Solution:
    def addSpaces(self, s: str, spaces: list[int]) -> str:
        result = []
        space_set = set(spaces)
        
        for i, char in enumerate(s):
            if i in space_set:
                result.append(" ")
            result.append(char)
        
        return "".join(result)
